hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java

```diff
@@ -244,319 +1097,596 @@ void supportedContractModificationFunctionsResponseBodyTest(
+                        .get()),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10000L * 100_000_000L));
+
+        // Then
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createTokenWithInvalidDecimalsFails() {
+        // Given
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+
+        // When
+        final var function = contract.send_createFungibleTokenExternal(
+                convertTokenEntityToHederaToken(
+                        domainBuilder.token().customize(t -> t.decimals(-1)).get()),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10000L * 100_000_000L));
+
+        // Then
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createTokenWithInvalidInitialSupplyFails() {
+        // Given
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+
+        // When
+        final var function = contract.send_createFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.initialSupply(-1L))
+                        .get()),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10000L * 100_000_000L));
+
+        // Then
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createTokenWithInvalidInitialSupplyGreaterThanMaxSupplyFails() {
+        // Given
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+
+        // When
+        final var function = contract.send_createFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.initialSupply(10_000_001L))
+                        .get()),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10L),
+                BigInteger.valueOf(10000L * 100_000_000L));
+
+        // Then
+        assertThatThrownBy(function::send).isInstanceOf(MirrorEvmTransactionException.class);
+    }
+
+    @Test
+    void createNftWithNoSupplyKeyFails() {
+        // Given
+        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
+
+        // When
+        final var function = contract.send_createNonFungibleTokenExternal(
+                convertTokenEntityToHederaToken(domainBuilder
+                        .token()
+                        .customize(t -> t.supplyKey(new byte[0]))
+                        .get()),
```

Comment:
DomainBuilder invocation and complex object setup should be performed in the given section. Same comment elsewhere.

```suggestion
        final var contract = testWeb3jService.deploy(ModificationPrecompileTestContract::deploy);
        final var token = domainBuilder.token().customize(t -> t.supplyKey(new byte[0])).get();
        final var hederaToken = convertTokenEntityToHederaToken(token);

        // When
        final var function = contract.send_createNonFungibleTokenExternal(hederaToken,```

Reply:
Will adapt in next PRs